### PR TITLE
Match user inheritance

### PR DESCRIPTION
**Pull request by inaod568**
Changes to match [the User inheritance PR](https://github.com/ScilifelabDataCentre/dds_web/pull/627)

### CODE REVIEW

File: dds_cli/__main__.py [hunk 1]
```diff
@@ -11,40 +11,40 @@
...
+import rich.progress
 import rich.prompt
-from rich import pretty
-from rich.progress import Progress, BarColumn
+import simplejson
+import questionary
 
 # Own modules
 import dds_cli
+import dds_cli.admin
```
Comment (alneberg) [inline dds_cli/__main__.py:28]: This causes an error on the CLI, I assume its something you'll add in the next PR. But probably good to delete it here so it's not breaking the cli

File: dds_cli/__main__.py [hunk 2]
```diff
@@ -11,40 +11,40 @@
...
 import dds_cli.exceptions as exc
 import dds_cli.data_getter
 import dds_cli.data_lister
 import dds_cli.data_putter
 import dds_cli.data_remover
 import dds_cli.directory
-import dds_cli.timestamp
+import dds_cli.option_classes
```
Comment (alneberg) [inline dds_cli/__main__.py:35]: Same here I think.

File: dds_cli/base.py [hunk 3]
```diff
@@ -84,8 +91,8 @@ def __init__(
         LOG.debug(f"Method: {self.method}, Project: {self.project}")
         # Project access only required if trying to upload, download or list
         # files within project
-        if self.method in ["put", "get"] or (
-            self.method in ["ls", "rm"] and self.project is not None
+        if self.method in DDS_PROJ_REQUIRED_METHODS or (
+            self.method in DDS_PROJ_NOT_REQUIRED_METHODS and self.project
```
[the remark this answers is not in the document]
Comment (inaod568) [reply dds_cli/base.py:95]: Sorry, I don't follow. Do you mean that the `DDS_PROJ_NOT_REQUIRED_METHODS` is not defined (it's in `__init__.py` too)?

File: dds_cli/data_putter.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
 import itertools
 import logging
 import pathlib
+import json
```
[the remark this answers is not in the document]
Comment (inaod568) [reply dds_cli/data_putter.py:12]: Doesn't seem like it, no! Removing

**Review by alneberg — APPROVED** [review]
Looks good!

**Pull request merged**